pkg/user/interactors/create_session.go

```diff
@@ -0,0 +1,57 @@
+package interactors
+
+import (
+	"github.com/CESARBR/knot-babeltower/pkg/cache"
+	"github.com/CESARBR/knot-babeltower/pkg/jwt"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/delivery/http"
+)
+
+// CreateSession is a use case operation that creates a session for the user.
+// This session is represented by a random ID, which will be further used for
+// publishing thing's data.
+type CreateSession struct {
+	thingsProxy  http.ThingProxy
+	generator    Generator
+	sessionStore cache.SessionStore
+}
+
+// NewCreateSession creates a new CreateSession instance by receiving its dependencies.
+func NewCreateSession(
+	thingsProxy http.ThingProxy,
+	generator Generator,
+	sessionStore cache.SessionStore,
+) *CreateSession {
+	return &CreateSession{thingsProxy, generator, sessionStore}
+}
+
+// Execute creates a new user session by receiving the authorization token as parameter.
+// This function will verify if a session already exists for the users. If not, it will
+// create a new session ID and save it in some caching storage.
+func (cs *CreateSession) Execute(authorization string) (string, error) {
+	email, err := jwt.GetEmail(authorization)
+	if err != nil {
+		return "", ErrParseToken
+	}
```

Comment:
Doesn't this "override" the error propagated by [jwt](https://github.com/CESARBR/knot-babeltower/pull/100/commits/b4b9efac8737e98a1dfd7ce3a3d9d0356f0c69ca#diff-557b22a4903a21e641140ed9c11e221b8a3d461b8b4b55553d1c17562a255219R30)?

Considering that `jwt.GetEmail` is another package, I'd suggest not ignoring the original error message and instead add context when propagating the error upwards:

```suggestion
	email, err := jwt.GetEmail(authorization)
	if err != nil {
		return "", fmt.Errorf("failed to create user: %v", err)
		// return "", fmt.Errorf("invalid format: %v", err)
		// return "", fmt.Errorf("%v: %v", SomeOtherVarWithErr, err)
	}
```

Reply:
Agreed. Done.

Reply:
Just to complement, remember to use the `%w` in the previously error, this is the guarantee that the `jwt.GetEmail` returned error could be checked with `error.is` function